core/arch/x86/decode_table.c

```diff
@@ -6341,113 +6762,113 @@ const instr_info_t third_byte_3a[] = {
  */
 const instr_info_t vex_W_extensions[][2] = {
   {    /* vex_W_ext 0 */
-    {OP_vfmadd132ps,0x66389818,"vfmadd132ps",Vvs,xx,Hvs,Wvs,Vvs,mrm|vex|reqp,x,END_LIST},
-    {OP_vfmadd132pd,0x66389858,"vfmadd132pd",Vvs,xx,Hvs,Wvs,Vvs,mrm|vex|reqp,x,END_LIST},
+    {OP_vfmadd132ps,0x66389818,"vfmadd132ps",Vvs,xx,Hvs,Wvs,Vvs,mrm|vex,x,tevexw[62][0]},
```

Comment:
Because the instructions are now multiplexed through the prefix_extensions and reqp is not needed. If you add it, the instruction fails to decode, because in read_prefix_ext,

```
if (di->data_prefix &&
        /* Don't remove it if the entry doesn't list 0x66:                                                                                                                                                                                                                                
         * e.g., OP_bsr (i#1118).                                                                                                                                                                                                                                                         
         */
        (info->opcode >> 24) == DATA_PREFIX_OPCODE)
        di->data_prefix = false;
```

I haven't debugged why or for which instructions this path is needed, but REQUIRES_PREFIX is used to skip the prefix extensions table, and we're not skipping it anymore.

Reply:
That said, I could have routed them through the e_vex_extensions table, but I didn't want to re-factor everything if there will be more same-opcode instructions using a different prefix. It's not a big overhead.

Reply:
> If you add it, the instruction fails to decode

?  But it was already there and it successfully decoded.

> REQUIRES_PREFIX is used to skip the prefix extensions table, and we're not skipping it anymore.

It sure looks like we're skipping it: for `OP_vfmadd132ps,0x66389818` we go to second_byte[], then third_byte_38[], then vex_W_extensions[]: I don't see any prefix_extensions.

Reply:
No, the decoder now goes through prefix_extensions. e.g. for vfmadd132ps:

{PREFIX_EXT,  0x389818, "(prefix ext 160)", xx, xx, xx, xx, xx, mrm, x, 160},/*60*/

And from there to vex_w_- and evex_w_extensions. It successfully decoded before, because before it skipped prefix_extensions (routed from third_byte_38).

Reply:
> I did not do any comparison vs other decoders -- assuming that has already been done.

That has been done, indeed.